### PR TITLE
add linworks fave60a

### PR DESCRIPTION
**Pull request by duilchoi**
add fave60a

## Description

hi.
i added linworks fave60a.
please review.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by duilchoi** [comment]
info.json file updated!
thank you.drashna!

File: keyboards/linworks/fave60a/info.json [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        "rgblight": false
+    },
+    "matrix_pins": {
+        "cols": ["B10", "B2", "B1", "B0", "A7", "A6", "A5", "A4", "A3", "A2", "A1", "A0", "C15", "C14", "C13"],
+        "rows": ["A8", "B15", "B14", "B8", "B9"]
+    },
+    "matrix_size": {
+        "cols": 15,
+        "rows": 5
+    },
```
Comment (drashna) [inline keyboards/linworks/fave60a/info.json:31]:
```suggestion
```

File: keyboards/linworks/fave60a/info.json [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    "mouse_key": {
+        "enabled": false
+    },
+    "platform": "STM32",
+    "processor": "STM32F072",
+    "processor_type": "arm",
+    "protocol": "ChibiOS",
```
Comment (drashna) [inline keyboards/linworks/fave60a/info.json:38]:
```suggestion
    "processor": "STM32F072",
```

File: keyboards/linworks/fave60a/info.json [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    },
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x000C",
+        "vid": "0x4C58"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "c_macro": true,
```
Comment (drashna) [inline keyboards/linworks/fave60a/info.json:143]:
```suggestion
```

File: keyboards/linworks/fave60a/readme.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# linworks/fave60a
+
+![linworks/fave60a front](https://imgur.com/UeWR8Kl)
+![linworks/fave60a back](https://imgur.com/j3mu8LF)
```
Comment (drashna) [inline keyboards/linworks/fave60a/readme.md:4]:
```suggestion
![linworks/fave60a front](https://i.imgur.com/UeWR8Klh.png)
![linworks/fave60a back](https://i.imgur.com/j3mu8LFh.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/linworks/fave60a/info.json [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    "url": "",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x000C",
+        "vid": "0x4C58"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+                { "label": "Esc", "matrix": [0, 0], "w": 1, "x": 0, "y": 0 },
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/info.json:137]:
```suggestion
                { "label": "Esc", "matrix": [0, 0], "x": 0, "y": 0 },
```
etc...

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/info.json (hunk 5 above)
Comment (fauxpark) [reply keyboards/linworks/fave60a/info.json]: The rest of these still need to be done.

File: keyboards/linworks/fave60a/config.h [hunk 6]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 ziptyze (@ziptyze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+
+/* Define RGB */
+#define RGBLED_NUM 91
+#define RGB_MATRIX_LED_COUNT 91
+#define RGB_DI_PIN A10
```
Comment (drashna) [inline keyboards/linworks/fave60a/config.h:8]:
```suggestion
#define WS2812_DI_PIN A10
#define RGB_DI_PIN WS2812_DI_PIN
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/info.json [hunk 7]
```diff
@@ -0,0 +1,202 @@
+{
+    "manufacturer": "Lx3",
+    "keyboard_name": "fave60a",
+    "maintainer": "keydler",
+    "bootloader": "stm32-dfu",
+    "debounce": 5,
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/info.json:6]:
```suggestion
```

File: keyboards/linworks/fave60a/readme.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# linworks/fave60a
+
+![linworks/fave60a front](https://i.imgur.com/UeWR8Klh.png)
+![linworks/fave60a back](https://i.imgur.com/j3mu8LFh.png)
+
+*A short description of the keyboard/project*
+
+* Keyboard Maintainer: [keydler](https://github.com/duilchoi)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/readme.md:10]: Please fill these out.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/readme.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
+# linworks/fave60a
+
+![linworks/fave60a front](https://i.imgur.com/UeWR8Klh.png)
+![linworks/fave60a back](https://i.imgur.com/j3mu8LFh.png)
+
+*A short description of the keyboard/project*
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/readme.md:6]:
```suggestion
```

File: keyboards/linworks/fave60a/readme.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+The Fave60a PCB is a 60 keys keyboard designed by Lx3 (Linworks).
+Matrix and Under RGB supported.
+
+* Keyboard Maintainer: [keydler](https://github.com/duilchoi)
+* Hardware Supported: *The PCBs, controllers supported*
+* Hardware Availability: *Links to where you can find this hardware*
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/readme.md:12]: These two still need to be filled out.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/info.json [hunk 11]
```diff
@@ -0,0 +1,201 @@
+{
+    "manufacturer": "Lx3",
+    "keyboard_name": "fave60a",
+    "maintainer": "keydler",
+    "bootloader": "stm32-dfu",
+    "diode_direction": "COL2ROW",
+    "encoder": {
+        "enabled": false
+    },
```
Comment (drashna) [inline keyboards/linworks/fave60a/info.json:9]:
```suggestion
```

File: keyboards/linworks/fave60a/info.json [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+        "rgb_matrix": true,
+        "rgblight": false
+    },
+    "matrix_pins": {
+        "cols": ["B10", "B2", "B1", "B0", "A7", "A6", "A5", "A4", "A3", "A2", "A1", "A0", "C15", "C14", "C13"],
+        "rows": ["A8", "B15", "B14", "B8", "B9"]
+    },
+    "mouse_key": {
+        "enabled": false
+    },
```
Comment (drashna) [inline keyboards/linworks/fave60a/info.json:29]:
```suggestion
```

File: keyboards/linworks/fave60a/info.json [hunk 13]
```diff
@@ -0,0 +1,201 @@
...
+    "features": {
+        "audio": false,
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
+        "extrakey": true,
+        "mousekey": false,
```
Comment (drashna) [inline keyboards/linworks/fave60a/info.json:18]:
```suggestion
        "encoder": true,
        "extrakey": true,
        "mousekey": true,
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/config.h [hunk 14]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 ziptyze (@ziptyze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+
+/* Define RGB */
+#define RGBLED_NUM 91
+#define RGB_MATRIX_LED_COUNT 91
+#define WS2812_DI_PIN A10
+#define RGB_DI_PIN WS2812_DI_PIN
```
Comment (drashna) [inline keyboards/linworks/fave60a/config.h:9]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/config.h [hunk 15]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 ziptyze (@ziptyze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+
+/* Define RGB */
+#define RGBLED_NUM 91
+#define RGB_MATRIX_LED_COUNT 91
+#define WS2812_DI_PIN A10
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/config.h:8]:
```suggestion
```

File: keyboards/linworks/fave60a/info.json [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
+    },
+    "matrix_pins": {
+        "cols": ["B10", "B2", "B1", "B0", "A7", "A6", "A5", "A4", "A3", "A2", "A1", "A0", "C15", "C14", "C13"],
+        "rows": ["A8", "B15", "B14", "B8", "B9"]
+    },
+    "processor": "STM32F072",
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/info.json:24]:
```suggestion
    "processor": "STM32F072",
    "ws2812": {
        "pin": "A10"
    },
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/config.h [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+#define ENABLE_RGB_MATRIX_HUE_BREATHING
+#define ENABLE_RGB_MATRIX_HUE_PENDULUM
+#define ENABLE_RGB_MATRIX_HUE_WAVE
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN
+// The PIXEL_FRACTAL effect does not work properly when the matrix layout is
+// different from the physical layout; it also has problems when underglow
+// LEDs are present, or when multiple LEDs are associated with the same key.
+#undef ENABLE_RGB_MATRIX_PIXEL_FRACTAL
+
+// Framebuffer effects; can be enabled only if RGB_MATRIX_FRAMEBUFFER_EFFECTS
```
Comment (drashna) [inline keyboards/linworks/fave60a/config.h:50]: Undefining `RGB_MATRIX_FRAMEBUFFER_EFFECTS` will effectively disable these too.

But if these aren't enabled, they don't need to be undef'ed

File: keyboards/linworks/fave60a/config.h [hunk 18]
```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 ziptyze (@ziptyze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+
+/* Define RGB */
+#define RGBLED_NUM 91
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/config.h:6]:
```suggestion
```

File: keyboards/linworks/fave60a/config.h [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
+#endif
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/config.h:71]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED
#define RGB_MATRIX_KEYPRESSES 
#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 120

#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL

#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_PIXEL_RAIN
// The PIXEL_FRACTAL effect does not work properly when the matrix layout is
// different from the physical layout; it also has problems when underglow
// LEDs are present, or when multiple LEDs are associated with the same key.
#undef ENABLE_RGB_MATRIX_PIXEL_FRACTAL

// Framebuffer effects; can be enabled only if RGB_MATRIX_FRAMEBUFFER_EFFECTS
// is defined.  Both of these effects currently don't work properly when the
// key matrix does not match the physical layout, so they are disabled.
#undef ENABLE_RGB_MATRIX_TYPING_HEATMAP
#undef ENABLE_RGB_MATRIX_DIGITAL_RAIN

// Reactive effects; can be enabled only if at least one of
// RGB_MATRIX_KEYPRESSES or RGB_MATRIX_KEYRELEASES is defined.
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```
These can be moved to info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/config.h [hunk 20]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 ziptyze (@ziptyze)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+
```
Comment (drashna) [inline keyboards/linworks/fave60a/config.h:4]:
```suggestion
#pragma once

```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/linworks/fave60a/rules.mk [hunk 21]
```diff
@@ -0,0 +1,2 @@
+# Enter lower-power sleep mode when on the ChibiOS idle thread
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
```
Comment (fauxpark) [inline keyboards/linworks/fave60a/rules.mk:2]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/linworks/fave60a/info.json [hunk 22]
```diff
@@ -0,0 +1,198 @@
...
+        "backlight": false,
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
```
Comment (waffle87) [inline keyboards/linworks/fave60a/info.json:18]:
```suggestion
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgb_matrix": true
```

**Review by waffle87 — APPROVED** [review]

**Pull request merged**